eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -265,19 +269,21 @@ private XElement GetPackageMappingsElementForSource(string packageSource)
             foreach (string packagePattern in allSourcesPackages[packageSource])
             {
                 // Add all packages from current source-built sources.
-                // For previously source-built and prebuilt sources add only packages
-                // where version does not exist in current source-built sources.
-                if (isCurrentSourceBuiltSource || !currentPackages.ContainsKey(packagePattern))
+                if (isCurrentSourceBuiltSource)
                 {
                     pkgSrc.Add(new XElement("package", new XAttribute("pattern", packagePattern)));
                 }
                 else if (packageSource.Equals(PreviouslySourceBuiltSourceName))
                 {
-                    AddPackageSourceMappingIfPackageVersionsNotInCurrentPackages(pkgSrc, packagePattern, previouslySourceBuiltPackages);
+                    AddPackageSourceMappingIfPackageVersionsNotInSources(pkgSrc, packagePattern, previouslySourceBuiltPackages, checkSharedComponents: true);
                 }
                 else if (packageSource.Equals(PrebuiltSourceName))
                 {
-                    AddPackageSourceMappingIfPackageVersionsNotInCurrentPackages(pkgSrc, packagePattern, prebuiltPackages);
+                    AddPackageSourceMappingIfPackageVersionsNotInSources(pkgSrc, packagePattern, prebuiltPackages, checkSharedComponents: true);
+                }
+                else if (packageSource.Equals(SharedComponentsSourceName))
+                {
+                    AddPackageSourceMappingIfPackageVersionsNotInSources(pkgSrc, packagePattern, sharedComponentsPackages, checkSharedComponents: false);
```

Comment:
Nit - maybe we could pass `packageSource` to `AddPackageSourceMappingIfPackageVersionsNotInSources` and decide if we check shared components based on that value, in the method.